### DISCUSSION TITLE
Adding custom .img files for custom flash layout to the bootloader folder

### DISCUSSION THREAD

**Discussion by goetzdd** [OE4T/tegra-demo-distro#131] [Q&A]
Ok, next question. I have some img files I want to use with the custom flash layout that I figured out how to use. Including setting the IMAGE_ROOTFS size smaller to be able to leave space for my new partitions. 

HINT, leave the "UDA" partition just after the APP_b, modify for size (3221225472 is what the default DATAFILE is (xxx.dataimg)) and change the allocation attribute to 0x08, then add your partitions, with your last one using 0x808 for the allocation attribute. Otherwise you have to track down where the mender layout sets the partitions or you'll be wondering why mender service and other things fail.

Anyway, how to get the xxx.img files I want to use into the proper place?

I tried a tegra-bootfiles_%.bbappend 

`FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"

SRC_URI_append = " \
    file://xxx.img \
"

BOOTBINS_append = " \
    xxx.img \
"`

but it ends up failing:

'install: cannot stat '/home/dgoetz/yocto/src/tegra-demo-distro/jetson-tx2-asg929/tmp/work-shared/L4T-tegra-32.5.1-r0/Linux_for_Tegra/bootloader/xxx.img': No such file or directory'

I think maybe there is two places I need a bbappend, one to get the files into the bootloader folder, and the above to append the files to BOOTBINS, I'm guessing one of the other recipes gets the source (to which I'll need to append my img files) while bootfiles (well obviously now that I look at it more) moves files from bootloader to tegraflash...

Pardon my yocto naivity is showing again - further digging maybe I need to do something (by something I mean I have no idea) with image_type_tegra.bbclass overriding tegraflash_custom_post() (how?) to dump the img files in the IMGDEPLOYDIR before the zip is created...??

Thanks much

**Comment by madisongh** [comment]
The answer depends on where the contents for the extra partitions comes from. If you look at how the `/data` partition image is created, you'll see the contents being installed in the main rootfs, then split off from there as the ext4 images are created.  If your extra partition images are created the same way, then, adding `tegraflash_custom_pre()` function to your image recipe to copy in the additional `.ext4` files from `${IMAGE_DEPLOY_DIR}` (or using `mksparse` to create sparseimage files from the `ext4` images).

If the images are created by some other recipe, you would still need the `tegraflash_custom_pre` function, but you would also need to to add a dependency so that those images get populated somewhere that they can be copied in as part of the tegraflash package creation process.  If they are created by image recipes, you could use `TEGRAFLASH_PKG_DEPENDS += "other-image-recipe:do_image_complete"`, for instance.

**Reply by goetzdd** [reply]
I really should dig more to understand how this works, but probably will just tweak the startup script I already have that disk checks and mounts them to format them first if necessary. They are simply blank to start for the most part anyway. Its an end around but I think I have more important steps to solve first.
